src/dtemplate.d

```diff
@@ -4453,6 +4443,8 @@ extern (C++) Type reliesOnTident(Type t, TemplateParameters* tparams = null, siz
 
     if (!t)
         return null;
+
+    assert(tparams);
```

Comment:
Now `reliesOnTident` is only used for template parameter analysis, so we can assert this.